src/Microsoft.Identity.Web/Resource/RequiredScopeFilter.cs

```diff
@@ -41,7 +43,15 @@ public void OnAuthorization(AuthorizationFilterContext context)
             {
                 throw new ArgumentNullException(nameof(context));
             }
-            else if (context.HttpContext.User == null || context.HttpContext.User.Claims == null || !context.HttpContext.User.Claims.Any())
+
+            GetEffectiveScopes(context);
+
+            ValidateEffectiveScopes(context);
+        }
+
+        internal void ValidateEffectiveScopes(AuthorizationFilterContext context)
+        {
+            if (context.HttpContext.User == null || context.HttpContext.User.Claims == null || !context.HttpContext.User.Claims.Any())
```

Comment:
I think we'd want to add a test that `_effectiveAcceptedScopes` is not **null** (and throw if that's the case with an explicit exception telling the developer that they need provide either `RequiredScopesSettings` or `RequiredScopesSettings` in the `RequiredScope` attribute.

Reply:
fixed. but let's discuss.

---
In reply to: [573122624](https://github.com/AzureAD/microsoft-identity-web/pull/942#discussion_r573122624) [](ancestors = 573122624)